app/models/user.rb

```diff
@@ -3,4 +3,17 @@ class User < ApplicationRecord
 
   validates :email, presence: true, uniqueness: true
   validates :first_name, :last_name, presence: true
+  validate :password_correct
+  validate :email_correct
```

Comment:
You provide good solution, but i suggest use rails validations:
```suggestion
  validates :password, length: { minimum: 6 }
  validates :email, presence: true, uniqueness: true,
                    format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
```
You doesn't need in different email validator for this case, you can add format validation on your `validates :email, presence: true, uniqueness: true`
About password: use simple password field. If you want throw any custom error, you can add this in `validates`. Check how it works